File: C++/Maximum Sum Problem.cpp
class Solution{
    public:
        int maxSum(int n){
            return n==0 ? 0 : max(max(n/2,maxSum(n/2)) + max(n/3,maxSum(n/3)) + max(n/4,maxSum(n/4)),n);
        }
};

File: JAVA/Maximum Sum Problem.java
class Solution
{
    public int maxSum(int n) 
    { 
        if(n==0||n==1||n==2||n==3) return n;
       int a=(int)n/2;
       int b=(int)n/3;
       int c=(int)n/4;
       int d=a+b+c;
       a=maxSum(a);
       b=maxSum(b);
       c=maxSum(c);
       return Math.max(a+b+c,Math.max(n,d));
    } 
}
